native_client/ctcdecode/path_trie.cpp

```diff
@@ -102,20 +99,18 @@ PathTrie* PathTrie::get_path_trie(unsigned int new_char, unsigned int new_timest
   }
 }
 
-void PathTrie::get_path_vec(std::vector<unsigned int>& output, std::vector<unsigned int>& timesteps) {
-  // Recursive call: recurse back until stop condition, then append data in
-  // correct order as we walk back down the stack in the lines below.
-  if (parent != nullptr) {
-    parent->get_path_vec(output, timesteps);
+std::vector<unsigned int> PathTrie::get_path_vec() {
+  if (parent == nullptr) {
+	return std::vector<unsigned int>{};
   }
+  std::vector<unsigned int> output_tokens=parent->get_path_vec();
   if (character != ROOT_) {
-    output.push_back(character);
-    timesteps.push_back(timestep);
+  	output_tokens.push_back(character);
   }
+  return output_tokens;
 }
```

Comment:
This change is causing a new vector to be constructed at each step in the recursion chain just to hold a single character. Taking the output vector as a parameter makes callers a bit more awkward but avoids this entirely, so please keep it like that.

Reply:
Copy elision should cause only one vector to be used. Copy elision is even guaranteed by the C++ standard since C++20.
Using return values to return values is more natural.
However, both way would work, so if you still prefer the old way, I will change it.

Reply:
I have not had a careful review, but please keep in mind some of that might be compiled as C++11 or C++14

Reply:
After checking, I see that it is actually guaranteed since C++17, but not with named objects. 
I don't expect this change to make any significant difference even with C++11, and I still believe it is better to return by value.
However, it is completely unrelated to this PR, and you both seem to prefer the way it was done before, so I have reverted this unnecessary change.